js/run_js_binary.bzl

```diff
@@ -0,0 +1,146 @@
+"""Runs a binary as a build action. This rule does not require Bash (unlike native.genrule()).
+
+This fork of bazel-skylib's run_binary adds directory output support and better makevar expansions.
+"""
+
+load("@aspect_bazel_lib//lib:run_binary.bzl", _run_binary = "run_binary")
+load("@aspect_bazel_lib//lib:copy_to_bin.bzl", _copy_to_bin = "copy_to_bin")
+load("@bazel_skylib//lib:dicts.bzl", "dicts")
+
+def run_js_binary(
+        name,
+        tool,
+        env = {},
+        srcs = [],
+        output_dir = False,
+        outs = [],
+        args = [],
+        chdir = None,
+        stdout = None,
+        stderr = None,
+        exit_code_out = None,
+        silent_on_success = True,
+        copy_srcs_to_bin = True,
+        **kwargs):
+    """Wrapper around @aspect_bazel_lib run_binary that adds convienence attributes for using a js_binary tool.
+
+    This rule does not require Bash `native.genrule`.
+
+    Args:
+        name: Target name
+        tool: The tool to run in the action.
```

Comment:
Consider renaming to binary to be more in line with the doc string and self documenting

Reply:
Hmmm. That diverges from the upstream `run_binary` which this is meant to wrap so might be more confusing that way. Not sure. Wtdy @alexeagle .

It would be good to be able to assert that the tool is indeed a js_binary as well.

Reply:
yeah we should stay similar to run_binary.

This isn't JS-specific really, except the placeholder naming

Reply:
we'll use this wrapper for the generated "bin" build actions